gdal/frmts/netcdf/netcdflayer.cpp

```diff
@@ -2370,9 +2594,19 @@ OGRErr netCDFLayer::CreateField(OGRFieldDefn *poFieldDefn, int /* bApproxOK */)
             nDimCount = 2;
             int anDims[2] = {nMainDimId, nSecDimId};
             nType = NC_CHAR;
-            status = nc_def_var(m_nLayerCDFId, pszVarName, nType, 2, anDims,
-                                &nVarID);
+
+            if(!m_bLegacyCreateMode)
```

Comment:
see above remarks. could be merged in a single code path with a passthrough vcdf implementation